discoverer/src/main/java/eu/seaclouds/platform/discoverer/api/DiscovererAPI.java

```diff
@@ -0,0 +1,312 @@
+ *    limitations under the License.
+ */
+
+package eu.seaclouds.platform.discoverer.api;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import eu.seaclouds.platform.discoverer.core.Discoverer;
+import eu.seaclouds.platform.discoverer.core.Offering;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.Set;
+
+@Path("/")
+public class DiscovererAPI {
+
+    private final Discoverer discoverer;
+    private org.json.simple.parser.JSONParser JSONParser;
+    static Logger log = LoggerFactory.getLogger(DiscovererAPI.class);
+
+    public DiscovererAPI(Discoverer discoverer) {
+        this.discoverer = discoverer;
+        this.JSONParser = new JSONParser();
+    }
+
+    @POST
+    @Path("/delete")
+    @Produces(MediaType.APPLICATION_JSON)
+    public DeleteRepresentation deleteOfferingById(@QueryParam("oid") String offerId)
+            throws IOException {
+
+        ArrayList<String> removedOfferingIds = new ArrayList<>();
+
+        /* if the offering id is valid and the discoverer is able to remove the offering associated
+         * then the offering id is added to the list of removed offering  */
+        if(this.discoverer.removeOffering(offerId)) {
+            removedOfferingIds.add(offerId);
+        }
+
+        return new DeleteRepresentation(removedOfferingIds);
+    }
+
+    @GET
+    @Path("/fetch_all")
+    @Produces(MediaType.APPLICATION_JSON)
+    public FetchAllRepresentation getAllOfferings() throws IOException {
+
+        /* collecting all the ids within the repository */
+        ArrayList<String> ids = new ArrayList<String>();
+        Collection<String> offeringIds = discoverer.getAllOfferingIds();
+        for (String offeringId : offeringIds) {
+            ids.add(offeringId);
+        }
+
+        Offering offering = this.discoverer.fetchOffer("all");
+        String offeringTOSCA = "";
+
+        if(offering != null) {
+            offeringTOSCA = offering.toscaString;
+        }
+
+        if (offeringTOSCA.isEmpty()) {
+            ids.clear();
+        }
+
+        return new FetchAllRepresentation(ids, offeringTOSCA);
+    }
+
+    @GET
+    @Path("/fetch")
+    @Produces(MediaType.APPLICATION_JSON)
+    public ArrayList<String> getOfferingIds()
+            throws IOException {
+        /* collecting all the ids within the repository */
+
+        ArrayList<String> ids = new ArrayList<String>();
+        Collection<String> offeringIds = discoverer.getAllOfferingIds();
+        for (String offeringId : offeringIds) {
+            ids.add(offeringId);
+        }
+
+        return ids;
+    }
+
+    @POST
+    @Path("/fetch")
```

Comment:
I guess, something like this would be a little cleaner and easier to read, test and maintain
```
  @GET
  @Path("{id}")
  public Offering get(@PathParam("id") String id) {
         FindIterable<Document> cursor = this.offeringsCollection.find(query);
         Offering offering =  Offering.fromDB(cursor.first());
         if (offering == null) {
                throw new WebApplicationException(Response.Status.NOT_FOUND);
    }
    return offering;
  }
```
notice, I've not compiled it so you may need to adjust it